xds/internal/testutils/e2e/clientresources.go

```diff
@@ -358,21 +358,25 @@ func DefaultCluster(clusterName, edsServiceName string, secLevel SecurityLevel)
 }
 
 // DefaultEndpoint returns a basic xds Endpoint resource.
-func DefaultEndpoint(clusterName string, host string, port uint32) *v3endpointpb.ClusterLoadAssignment {
+func DefaultEndpoint(clusterName string, host string, ports []uint32) *v3endpointpb.ClusterLoadAssignment {
+	var lbEndpoints []*v3endpointpb.LbEndpoint
+	for _, port := range ports {
+		lbEndpoints = append(lbEndpoints, &v3endpointpb.LbEndpoint{
+			HostIdentifier: &v3endpointpb.LbEndpoint_Endpoint{Endpoint: &v3endpointpb.Endpoint{
+				Address: &v3corepb.Address{Address: &v3corepb.Address_SocketAddress{
+					SocketAddress: &v3corepb.SocketAddress{
+						Protocol:      v3corepb.SocketAddress_TCP,
+						Address:       host,
+						PortSpecifier: &v3corepb.SocketAddress_PortValue{PortValue: uint32(port)}},
```

Comment:
Do we need the type cast here since `ports` is already a `[]uint32`?

Reply:
Deleted